data/transactions/application.go

```diff
@@ -46,6 +46,12 @@ const (
 	// can contain. Its value is verified against consensus parameters in
 	// TestEncodedAppTxnAllocationBounds
 	encodedMaxForeignAssets = 32
+
+	// encodedMaxBoxes sets the allocation bound for the maximum
+	// number of Boxes that a transaction decoded off of the wire
+	// can contain. Its value is verified against consensus parameters in
+	// TestEncodedAppTxnAllocationBounds
+	encodedMaxBoxes = 32
```

Comment:
Why is this 32 and not 8? same q on the other foreign refs for that matter

Reply:
@barnjamin It's a fair question and I think I don't know either.  Sharing what I see and perhaps someone else can chime in.

* Via git blame, it _appears_ the initial value = 32 too:  https://github.com/algorand/go-algorand/blame/5d55a675688b92348a43c6b49ee1bf76faa41d85/data/transactions/application.go.  It's unclear to me how the initial value is configured.  I presume `encodedMaxBoxes` follows the prevailing convention.
* Since `TestEncodedAppTxnAllocationBounds` enforces the consensus parameter + alloc bounds validity, I don't see a clear rationale for configuring larger than needed values.

Reply:
I just copied the value used above for assets and apps, for which the same arguments could be made, in order to be consistent and stop people from wondering why they were different.  They could all be shrunk, I suppose.